consumer/consumer_storage_tornado.py

```diff
@@ -32,6 +32,34 @@ def decode(msg):
     msg = pickle.loads(msg)
     return msg
 
+def Bbox(text_file):
+    #add the IOU and hardcode locations
+
+    df = pd.DataFrame(text_file['predictions'])
+    df1 = pd.DataFrame(dict(df['boundingBox'])).T
+    df1['right'] = df1['left'] + df1['width']
+    df1['bottom'] = df1['top'] - df1['height']
+    df = df1.merge(df['probability'], right_index=True, left_index=True)
+    left_most = min(df['left'])
+    right_most = max(df['left']) + df.loc[df['left'] == max(df['left']),'width'].iloc[0]
+    bottom_most = min(df['top'])
+    top_most = max(df['top']) + df.loc[df['top'] == max(df['top']),'height'].iloc[0]
+    height_avg = df['height'].mean()
+    width_avg = df['width'].mean()
+    nrows = int((top_most - bottom_most)/height_avg)
+    ncols = int((right_most - left_most)/width_avg)
+    mat = np.zeros((nrows, ncols))
+    for i in range(nrows):
+        for j in range(ncols):
+            top_n = top_most - i*(height_avg)
+            bottom_n = top_most - (i+1)*height_avg
+            left_n = left_most + j*width_avg
+            right_n = left_most + (j+1)*width_avg
+            mat[i][j] = max(df['probability '].loc[(((df['top'] <= top_n) & (df['top'] >= bottom_n))
+                                                    | ((df['bottom'] <= top_n) & (df['bottom'] >= bottom_n)))
+                                                   | (((df['right'] <= right_n) & (df['right'] >= left_n))
+                                                      | ((df['left'] <= right_n) & (df['left'] >= left_n)))])
+    return mat
```

Comment:
Since we are using tornado, we need to return something that can be serialized and arrays cannot be. can you change the return statement to mat.tolist()

Reply:
I am not sure if that is what Kevin wants. He wanted a 2D array for his input, will converting into list loose its dimensionality?

Reply:
Unfortunately, arrays cannot be passed/returned by tornado for requests made. @kgao this is an example of the output after the tolist method. [[0.0, 0.0, 0.0, 0.0], [0.0, 0.0, 0.0, 0.0], [0.0, 0.0, 0.0, 0.0]]. This should work for you right?

Reply:
It is already in this form. I guess you mean it shouldn't be in numpy array form but rather in python list. Am I right?

Reply:
yes - it's a python list now